lib/open_project/authentication.rb

```diff
@@ -0,0 +1,47 @@
+require 'open_project/authentication/manager'
+
+module OpenProject
+  ##
+  # OpenProject uses Warden strategies for request authentication.
+  module Authentication
+    class << self
+      ##
+      # Updates the used warden strategies for a given scope. The strategies will be tried
+      # in the order they are set here. Plugins can call this to add or remove strategies.
+      # For available scopes please refer to `OpenProject::Authentication::Scope`.
+      #
+      # @param [Symbol] scope The scope for which to update the used warden strategies.
+      # @param [Boolean] store Indicates whether the user should be stored in the session
+      #                        for this scope.
```

Comment:
The point is that as a plugin developer I may just want to touch the strategies (add one, remove one, reorder) while not caring about the persistence of the logged in user. Do you want me to mention that in the comment?